resolver.go

```diff
@@ -184,7 +189,14 @@ func createRespMsg(query *dns.Msg) *dns.Msg {
 }
 
 func (r *resolver) handleIPQuery(name string, query *dns.Msg, ipType int) (*dns.Msg, error) {
-	addr, ipv6Miss := r.sb.ResolveName(name, ipType)
+	var addr []net.IP
+	var ipv6Miss bool
+	if r.sb != nil {
+		addr, ipv6Miss = r.sb.ResolveName(name, ipType)
+	} else if r.network != nil {
```

Comment:
We can abstract out this difference in a generic way; lets add a flag `noProxy` to the resolver structure which is a config option passed in the NewResolver call. It will be `false` in the *nix version of startResolver and `true` for Windows.

And in the ServeDNS we can check this to return `ServerFailure`.